doc/user/content/ingest-data/_index.md

```diff
@@ -0,0 +1,244 @@
+  snapshotting. Once the snapshotting operation is complete, you can downsize
+  the cluster to align with the steady-state ingestion.
+
+- See also [Best practices](#best-practices).
+
+{{% /tip %}}
+
+## Snapshotting
+
+Snapshotting refers to the initial population of a source in Materialize. When
+you create a new source, Materialize takes a snapshot of the data from the
+upstream (i.e., external) system at a given offset and loads it into
+Materialize.
+
+For the offset, Materialize chooses the latest available offset. For the
+available offset, Materialize uses:
+
+- Log Sequence Number (LSN) for PostgreSQL sources.
+
+- The number of transactions committed across all servers in the cluster for
+  MySQL sources.
+
+- Kafka message offset for Kafka sources.
+
+Materialize captures/commits a snapshot of all historical data up to that point.
+Snapshot is persisted in the storage layer, and all records in this initial load
+have the same ingestion timestamp.
+
+### Duration
+
+The snapshotting operation duration depends on the snapshot dataset size and the
+size of the Materialize cluster that is hosting the source. For very large
+sources that contain hundreds of GB of data, it can take up to an hour or even
+several hours to complete.
+
+{{% tip %}}
+
+- If possible, schedule creating new sources during off-peak hours to mitigate
+  the impact of snapshotting on both the upstream system and the Materialize
+  cluster.
+
+- Consider using a larger cluster size during snapshotting. Once the
+  snapshottingis complete, you can downsize the cluster to align with the volume
+  of changes being replicated from your upstream in steady-state.
+
+- See also [Best practices](#best-practices).
+
+{{% /tip %}}
+
+If you create your source from the Materialize Console, the overview page for
+the source displays the snapshotting progress. Alternatively, you can run a
+query to monitor its progress. See [Monitoring the snapshotting progress](/ingest-data/monitoring-data-ingestion/#monitoring-the-snapshotting-progress).
+
+### CPU and memory utilization
+
+Snapshotting may require more compute resources from the Materialize cluster
+than steady-state. If there are signs of resource exhaustion (that is, the
+cluster restarts because it ran out of memory), resize the cluster.
+
+{{% tip %}}
+Consider using a larger cluster size during snapshotting. Once the snapshotting
+operation is complete, you can downsize your source cluster to align with the
+volume of changes being replicated from your upstream in steady-state.
+{{% /tip %}}
+
+### Queries during snapshotting
+
+While a source is snapshotting, the source (and the associated subsources)
+cannot serve queries. That is, queries issued to the snapshotting source (and
+its subsources) will return after the snapshotting completes (unless the user
+breaks out of the query). If the user does not break out of the query, the
+returned query results will reflect the data from the snapshot.
```

Comment:
```suggestion
Because the initial snapshot is persisted atomically, you are **not able to
query the source until snapshotting is complete**. This means that queries
issued against (sub)sources undergoing snapshotting will hang until the
operation completes. Once the initial snapshot has been ingested, you can start
querying your (sub)source and Materialize will continue ingesting any new data
as it arrives, in real time.
```